### PR TITLE
Upgrade Java build to target JDK 17

### PR DESCRIPTION
**Pull request by pepone**
This PR upgrade the Java build to target Java 17, it also upgrade the Gradle version, and removes support for old Gradle versions.

### CODE REVIEW

File: java/build.gradle [hunk 1]
```diff
@@ -20,21 +18,15 @@ subprojects {
...
-    // Only enable infer module path with Java 9 and up
-    if(GradleVersion.current() >= GradleVersion.version('6.4')) {
-        java {
-            modularity.inferModulePath = targetJavaRelease.toInteger() > 8
-        }
+    java {
+        modularity.inferModulePath = true
+        // The corresponding -source and -target javac options are suppressed by gradle when
+        // --release is set through options.compileArgs (see above).
+        sourceCompatibility = "1.${targetJavaRelease}"
```
Comment (pepone) [inline java/build.gradle:28]: We need to set this with the Java plug-in, the old way of setting this is deprecated.

See: https://docs.gradle.org/8.7/userguide/upgrading_version_8.html#java_convention_deprecation

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/LiveDeployment/MetricsViewEditor.java [hunk 2]
```diff
@@ -862,13 +862,13 @@ public void addMetrics(com.zeroc.Ice.IceMX.Metrics m, long timestamp) {
         if (value == null) {
           Class c = getColumnClass(entry.getKey().intValue());
           if (c.equals(Integer.class)) {
```
Comment (pepone) [inline java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/LiveDeployment/MetricsViewEditor.java:864]: This fixes some Java 17 deprecation warnings.

**Review by bernardnormier — APPROVED** [review]

**Review by externl — APPROVED** [review]
Looks good!

File: java/gradlew.bat [hunk 3]
```diff
@@ -70,6 +70,7 @@ goto fail
 
 set CLASSPATH=%APP_HOME%\gradle\wrapper\gradle-wrapper.jar
 
+
```
Comment (InsertCreativityHere) [inline java/gradlew.bat:73]: Was this intentional?
It's fine if it was! Just wanted to make sure.

**Review by InsertCreativityHere — APPROVED** [review]

File: java/gradlew.bat (hunk 3 above)
Comment (externl) [reply java/gradlew.bat:73]: pretty sure these wrappers are generated by gradle and checked in

Comment (pepone) [reply java/gradlew.bat:73]: this is generated when you run the wrapper update comman. I just keep the generated file as is.

**Pull request merged**